qt_ui/windows/QDebriefingWindow.py

```diff
@@ -86,3 +88,9 @@ def __init__(self, debriefing: Debriefing):
         okay = QPushButton("Okay")
         okay.clicked.connect(self.close)
         layout.addWidget(okay)
+
+    def closeEvent(self, event: QCloseEvent) -> None:
+        super().closeEvent(event)
+        state = self.debriefing.game.check_win_loss()
```

Comment:
Why is this needed? Isn't this going to happen in `initialize_turn`? I'd rather the UI be responsible for as little game state management as possible.

Reply:
Deleted, refactoring to `game.process_win_loss`

Reply:
Ok now I remember why I did this. The check in "initialize turn" happens sooner, so if I emit the signal from there we never get to see the debriefing window because I force all windows to be closed except the main window.
I'll see if I can find a way around this...

Reply:
Turns out the debriefing window does actually show, but I can't seem to find a better spot to trigger the signal without guaranteeing it's only triggered once...

Reply:
> Why is this needed?

Two cases a game can end, after a cheat-capture and after a debriefing, this is the latter...

> Isn't this going to happen in `initialize_turn`?

Yes, but that would mean calling `GameUpdateSignal.get_instance().gameStateChanged`. I'm not a fan of that because that means UI stuff is making its way into the model. The main reason the check needs to happen in `initialize_turn` is to actually prevent turn initialization, because in an end-game situation that'll cause a bunch of exceptions.


> I'd rather the UI be responsible for as little game state management as possible.

Please elaborate because I really don't see how I'm "managing game state" by calling `check_win_loss`